### ISSUE TITLE
Version too old for updated clients

### ISSUE THREAD

**Issue by dhelgerson** [Thefrank/jellyfin-server-freebsd#90]
For a while now, jellyfin clients have been warning that this version of jellyfin is too old but the most recent update to the client causes it to refuse to connect, prompting a server update.

**Comment by Thefrank** [comment]
@dhelgerson The latest version from ports (`pkg install jellyfin`) should be new enough to work with current clients. I no longer provide a binary here as it is usually easier for people to use ports than manually grab/update from a Github repo.

**Comment by vgerpub** [comment]
I know it's an older thread, but I noticed that the later version (10.10.3_1) was pulled from the latest branch. Anyone know when it'll be back, or what the story is there? Now only the 10.9.6 version is available.

**Comment by Thefrank** [comment]
@vgerpub 
TL;DR summary: Jellyfin port needs to be updated for newer node version (and maybe dotnet port changes). No ETA but the port maintainer likely already knows about this.

From: https://www.freshports.org/multimedia/jellyfin/ it does appear missing outside of the quarterly branch.
There are two bugs filed against it too! https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=283307 and https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=283347

`{NODEJS_SUFFIX}` moved from Node20 to Node22 causing the Node version conflict (https://github.com/freebsd/freebsd-ports/commit/625260412b3217bce438f6c2e7eaae7bccb20cf0). Not related to those two bugs, there were large changes in the dotnet port: there is now a `dotnet-runtime`, `dotnet8` and `dotnet` can be installed side-by-side, and dotnet9 moved to using its in-tree `libunwind`.

**Comment by vgerpub** [comment]
@Thefrank 

Thanks for the reply and the update. I'll be keeping my eyes peeled for the next version.

**Comment by JeremyPaulCarroll** [comment]
> @dhelgerson The latest version from ports (`pkg install jellyfin`) should be new enough to work with current clients. I no longer provide a binary here as it is usually easier for people to use ports than manually grab/update from a Github repo.

Hi! Please forgive me if this isn't in any kind of standard for GitHub - this is the first time I'ver posted anything on GitHub, so please bear with me!

Building from updated ports, I get:

```
if [ ! -f /usr/ports/distfiles/jellyfin-10.10.3-node-modules.tar.gz ]; then  /bin/mkdir -p /usr/ports/multimedia/jellyfin/work/node-modules-cache;  cp -f -R /usr/ports/multimedia/jellyfin/files/packagejsons/* /usr/ports/multimedia/jellyfin/work/node-modules-cache;  cd /usr/ports/multimedia/jellyfin/work/node-modules-cache &&  /usr/bin/env HOME=/usr/ports/multimedia/jellyfin/work  npm ci --ignore-scripts --no-progress --no-audit --no-fund;  /usr/bin/find /usr/ports/multimedia/jellyfin/work/node-modules-cache -depth 1 -print |  /usr/bin/grep -v node_modules | /usr/bin/xargs /bin/rm -f -r;  /usr/bin/find /usr/ports/multimedia/jellyfin/work/node-modules-cache -type d -exec /bin/chmod 755 {} ';';  cd /usr/ports/multimedia/jellyfin/work/node-modules-cache &&  /usr/sbin/mtree -cbnSp node_modules | /usr/sbin/mtree -C | /usr/bin/sed  -e 's:time=[0-9.]*:time=61171200.000000000:'  -e 's:\([gu]id\)=[0-9]*:\1=0:g'  -e 's:flags=.*:flags=none:'  -e 's:^\.:./node_modules:' > node-modules-cache.mtree;  /usr/bin/tar -cz --options 'gzip:!timestamp'  -f /usr/ports/distfiles/jellyfin-10.10.3-node-modules.tar.gz  @node-modules-cache.mtree;  /bin/rm -f -r /usr/ports/multimedia/jellyfin/work;  fi
npm warn deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm warn deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm warn deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm warn deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm warn deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm warn deprecated @types/vfile-message@2.0.0: This is a stub types definition. vfile-message provides its own type definitions, so you do not need this installed.
npm warn deprecated @types/localforage@0.0.34: This is a stub types definition for localforage (https://github.com/localForage/localForage). localforage provides its own type definitions, so you don't need @types/localforage installed!
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated @stylelint/postcss-markdown@0.36.2: Use the original unforked package instead: postcss-markdown
npm warn deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
npm warn deprecated @stylelint/postcss-css-in-js@0.37.3: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.

added 1903 packages in 2m
npm notice
npm notice New major version of npm available! 10.8.3 -> 11.0.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.0.0
npm notice To update run: npm install -g npm@11.0.0
npm notice
=> jellyfin-10.10.3-node-modules.tar.gz doesn't seem to exist in /usr/ports/distfiles/.
===> /!\ Warning /!\
     The :prefetch group used for jellyfin-10.10.3-node-modules.tar.gz is missing
     from MASTER_SITES. Check for typos, or errors.
=> Attempting to fetch http://distcache.FreeBSD.org/ports-distfiles/jellyfin-10.10.3-node-modules.tar.gz
fetch: http://distcache.FreeBSD.org/ports-distfiles/jellyfin-10.10.3-node-modules.tar.gz: Not Found
=> Couldn't fetch it - please try to retrieve this
=> port manually into /usr/ports/distfiles/ and try again.
*** Error code 1

Stop.
make[1]: stopped in /usr/ports/multimedia/jellyfin
*** Error code 1

Stop.
make: stopped in /usr/ports/multimedia/jellyfin
```

**Comment by Thefrank** [comment]
@JeremyPaulCarroll what port management software are you using? `poudriere` should be generating that without issue.

**Comment by JeremyPaulCarroll** [comment]
> @JeremyPaulCarroll what port management software are you using? `poudriere` should be generating that without issue.

Ok, so it was an easy fix. If it's useful to anyone, here are the steps I followed:
`sudo nano /etc/pkg/FreeBSD.conf`
I changed the line:
`  url: "pkg+https://pkg.FreeBSD.org/${ABI}/quarterly",`
to:
`  url: "pkg+https://pkg.FreeBSD.org/${ABI}/latest",`
save and exit.
then stopped the jellyfin server:
`sudo service jellyfin stop`
then:
```
sudo pkg install multimedia/jellyfin
Updating FreeBSD repository catalogue...
pkg: Repository FreeBSD has a wrong packagesite, need to re-create database
Fetching meta.conf: 100%    178 B   0.2kB/s    00:01    
Fetching data.pkg: 100%    7 MiB   7.6MB/s    00:01    
Processing entries: 100%
FreeBSD repository update completed. 35944 packages processed.
All repositories are up to date.
The following 2 package(s) will be affected (of 0 checked):

New packages to be INSTALLED:
	libskiasharp: 2.88.3_1

Installed packages to be UPGRADED:
	jellyfin: 10.9.6 -> 10.10.3_1

Number of packages to be installed: 1
Number of packages to be upgraded: 1

75 MiB to be downloaded.

Proceed with this action? [y/N]: y
[1/2] Fetching jellyfin-10.10.3_1.pkg: 100%   73 MiB  25.6MB/s    00:03    
[2/2] Fetching libskiasharp-2.88.3_1.pkg: 100%    2 MiB   2.1MB/s    00:01    
Checking integrity... done (0 conflicting)
[1/2] Installing libskiasharp-2.88.3_1...
[1/2] Extracting libskiasharp-2.88.3_1: 100%
[2/2] Upgrading jellyfin from 10.9.6 to 10.10.3_1...
===> Creating groups
Using existing group 'jellyfin'
===> Creating users
Using existing user 'jellyfin'
[2/2] Extracting jellyfin-10.10.3_1: 100%
=====
Message from jellyfin-10.10.3_1:

--
Those upgrading, as a reminder:

This package installs a service file.
Enable it with "service jellyfin enable"
Start it with "service jellyfin start".

The service file uses daemon to restart jellyfin if it crashes.
The service file will also change the permissions so that the updater works.
If this behavior is unwanted you will need to edit the RC file manually and
remove the daemon and/or the permissions changes.

If you are running this in a jail please set "allow.mlock" or similar
for this jail otherwise the program will fail to start.

dotNET does not work well inside jails that are missing either a) VNET or 
b) ip6=inherit. The service file will try workaround any user misconfiguration 
but is not perfect.
sudo service jellyfin enable
jellyfin enabled in /etc/rc.conf
sudo service jellyfin start
Starting jellyfin.
sudo service jellyfin status
jellyfin is running as pid 84967.
```
And it was as simple as that. No issues, worked first time.

**Comment by Thefrank** [comment]
Glad it worked out. FreeBSD (and its jail creation) will usually default to `Quarterly` which can cause some confusion on why `pkg` is saying something is update to date despite other branches saying otherwise
